swarm/network/simulations/discovery/discovery_test.go

```diff
@@ -26,7 +26,9 @@ const serviceName = "discovery"
 func init() {
 	// register the discovery service which will run as a devp2p
 	// protocol when using the exec adapter
-	adapters.RegisterService(serviceName, discoveryService)
+	adapters.RegisterService(serviceName, func(id *adapters.NodeId) p2pnode.Service {
```

Comment:
It is fine for now yes, the code will panic on `init` if two packages try to register the same name, in which case we can just rename things.